### PR TITLE
[develop edition] GcsBatchFlow should be resilient to exceptions with null error message happening on attempt to execute GCS batch request [BW-411]

### PR DESCRIPTION
**Pull request by grsterin**

### CODE REVIEW

**Review by aednichols — APPROVED** [review]
tenor

**Review by kshakir — APPROVED** [review]
If codecov says everything is covered then LGTM

File: engine/src/test/scala/cromwell/engine/io/gcs/GcsBatchFlowSpec.scala [hunk 1]
```diff
@@ -26,16 +35,29 @@ class GcsBatchFlowSpec extends AnyFlatSpecLike with CromwellTimeoutSpec with Mat
...
+    implicit val ec: ExecutionContextExecutor = system.dispatcher
+    val gcsBatchFlow = new GcsBatchFlow(1, system.scheduler, null, "testAppName")
+
+    val mockGcsPath = GcsPath(CloudStorageFileSystem.forBucket("bucket").getPath("test"), any[com.google.api.services.storage.Storage], any[com.google.cloud.storage.Storage], anyString)
+    val gcsBatchCommandContext = GcsBatchCommandContext(GcsBatchCrc32Command(mockGcsPath), TestProbe().ref)
+    val recoverCommandPrivateMethod = PrivateMethod[PartialFunction[Throwable, Future[GcsBatchResponse[_]]]]('recoverCommand)
+    val partialFuncAcceptingThrowable = gcsBatchFlow invokePrivate recoverCommandPrivateMethod(gcsBatchCommandContext)
 
-    Set(new RuntimeException("random exception")) flatMap GcsBatchFlow.getReadForbiddenBucket shouldBe Set.empty
+    val futureRes = partialFuncAcceptingThrowable(new NullPointerException(null)) // no unhandled exceptions should thrown here
```
Comment (mcovarr) [inline engine/src/test/scala/cromwell/engine/io/gcs/GcsBatchFlowSpec.scala:60]:
```suggestion
    val futureRes = partialFuncAcceptingThrowable(new NullPointerException(null)) // no unhandled exceptions should be thrown here
```

File: engine/src/test/scala/cromwell/engine/io/gcs/GcsBatchFlowSpec.scala [hunk 2]
```diff
@@ -26,16 +35,29 @@ class GcsBatchFlowSpec extends AnyFlatSpecLike with CromwellTimeoutSpec with Mat
...
+    Set(new RuntimeException("random exception")) map (_.getMessage) flatMap GcsBatchFlow.getReadForbiddenBucket shouldBe Set.empty
+  }
+
+  "GcsBatchFlow" should "not throw unhandled exception and kill the thread when trying to recover from unretryable exception with null error message" in {
+    implicit val ec: ExecutionContextExecutor = system.dispatcher
+    val gcsBatchFlow = new GcsBatchFlow(1, system.scheduler, null, "testAppName")
+
+    val mockGcsPath = GcsPath(CloudStorageFileSystem.forBucket("bucket").getPath("test"), any[com.google.api.services.storage.Storage], any[com.google.cloud.storage.Storage], anyString)
+    val gcsBatchCommandContext = GcsBatchCommandContext(GcsBatchCrc32Command(mockGcsPath), TestProbe().ref)
+    val recoverCommandPrivateMethod = PrivateMethod[PartialFunction[Throwable, Future[GcsBatchResponse[_]]]]('recoverCommand)
```
Comment (mcovarr) [inline engine/src/test/scala/cromwell/engine/io/gcs/GcsBatchFlowSpec.scala:57]: this is magic

**Review by mcovarr — APPROVED** [review]

File: engine/src/test/scala/cromwell/engine/io/gcs/GcsBatchFlowSpec.scala (hunk 2 above)
Comment (grsterin) [reply engine/src/test/scala/cromwell/engine/io/gcs/GcsBatchFlowSpec.scala:57]: Yeah, I learned this technique recently from Mark's S3 PR

**Comment by cjllanwarne** [comment]
@grsterin LGTM. Reminder to also make a hotfix edition of this change

**Review by cjllanwarne — APPROVED** [review]

**Pull request merged**